pgfinder/pgio.py

```diff
@@ -334,3 +350,98 @@ def read_yaml(filename: Union[str, Path]) -> Dict:
+    intensity_var: str = "Intensity",
+) -> pd.DataFrame:
+    """Convert long to wide format based on user specified id."""
+    # Subset the variables that are to be kept in long format for subsequent merging, adding counters so we don't end
+    # up with duplicates from merging long with wide format data based on id var.
+    keep_columns = [
+        id,
+        "Ion count",
+        "Charge state",
+        "XIC start (min)",
+        "XIC end (min)",
+        "RT (min)",
+        "Obs (Da)",
+        "Theo (Da)",
+        "∆ppm",
+        structure_var,
+        intensity_var,
+    ]
+    keep_other = df[keep_columns].copy()
+    # keep_other.rename({intensity_var: intensity_var + " (All)"}, axis=1, inplace=True)
+    keep_other["match"] = keep_other.groupby(id).cumcount() + 1
+
+    # Subset the variables that need reshaping, adding counters
+    keep_reshape = [
+        id,
+        structure_var,
+        intensity_var,
+    ]
+    to_reshape = df[df[structure_var].notna()][keep_reshape]
+    to_reshape["match"] = to_reshape.groupby(id).cumcount() + 1
+
+    # Retain only those instances where there is an intensity_var, this removes secondary (tertiary or quarternay etc.)
+    # matches without the lowest detla-ppm. Need to track how many tied matches there are for tidying up.
+    to_reshape = to_reshape[to_reshape[intensity_var].notna()]
+    total_duplicate_matches = to_reshape["match"].max()
+    to_reshape["match"] = to_reshape["match"].apply(str)
+
+    # Reshape to wide _only_ instances where there are two or more candidate matches,
+    # rename columns, drop the uneeded duplicate intensity columns
+    wide_df = pd.pivot(to_reshape, index=id, values=[structure_var, intensity_var], columns="match")
+    wide_df.columns = [" ".join(col).strip() for col in wide_df.columns.values]
+    wide_df.reset_index(inplace=True)
+    for x in range(2, total_duplicate_matches + 1):
+        wide_df.drop(" ".join([intensity_var, str(x)]), axis=1, inplace=True)
+    wide_df["match"] = 1
+    to_concatenate = [x for x in wide_df.columns if structure_var in x]
+    wide_df["Inferred structure (consolidated)"] = wide_df[to_concatenate].apply(
+        lambda row: ",   ".join(row.values.astype(str)), axis=1
+    )
+    wide_df["Inferred structure (consolidated)"] = wide_df["Inferred structure (consolidated)"].str.replace(
+        ",   nan", ""
+    )
+    wide_df.rename({"Intensity 1": "Intensity (consolidated)"}, axis=1, inplace=True)
+    wide_df.drop(columns=to_concatenate, inplace=True)
+
+    # Merge with long format data
+    wide_df = keep_other.merge(wide_df, on=[id, "match"], how="outer")
+    wide_df.drop("match", axis=1, inplace=True)
+    # Remove instances where there are multiuple matches with the same delta-ppm
+    # Code commented, originally wasn't done, then was, now back to original
+    # wide_df.drop_duplicates(
+    #     subset=[
+    #         id,
+    #         "∆ppm",
+    #     ],
+    #     inplace=True,
+    # )
```

Comment:
This could go?

Reply:
I left this in deliberately because of the changes in what was required should they be reverted at a later stage.

Happy to remove and re-write it again though (or dig it out from the relevant commit).